include/evm_runtime/evm_contract.hpp

```diff
@@ -84,6 +84,10 @@ class [[eosio::contract]] evm_contract : public contract
    /// @return true if all garbage has been collected
    [[eosio::action]] bool gc(uint32_t max);
 
+   
+   [[eosio::action]] void call(eosio::name from, const bytes& to, uint128_t value, const bytes& data, uint64_t gas_limit);
+   [[eosio::action]] void admincall(const bytes& from, const bytes& to, uint128_t value, const bytes& data, uint64_t gas_limit);
```

Comment:
the value here would only be used for EOS for a valid EVM transaction, so it will never exceed uint128.
For consistency, i think we should either use bytes for both value and gas_limit or we just pick a large enough integer type.

Reply:
I guess I am okay with `uint128_t` used for the value. But it makes me realize we really need test cases to verify that `call` and `admincall` do work with positive `value`. In the test, choose a `value` that is large enough to not fit in a 64-bit integer (so I think 1000 EOS should be sufficient) and verify the correct value was transferred.

As for `gas_limit`, I believe it shouldn't be possible for a transaction to consume more than gas than can be represented in a `uint32_t` anyway.

Reply:
done, 50_ether is enough to hit the limit.